src/App.jsx

```diff
@@ -1,60 +1,150 @@
-import "./App.css"
+import { useState } from "react";
+import "./App.css";
 
 function App() {
+  const [display, setDisplay] = useState("0");
+  const [operation, setOperation] = useState("+");
+  const [secondDisplay, setSecondDisplay] = useState("0");
+  const [result, setResult] = useState("0");
+  const [storedValue, setStoredValue] = useState(null);
+
+  const handleNumberClick = (number) => {
+    if (number === "." && !display.includes(".")) {
+      setDisplay(display + number);
+    } else if (number !== "." && (display === "0" || display === "0.")) {
+      setDisplay(number);
+    } else if (number !== ".") {
+      setDisplay(display + number);
+    }
+  };
+
+  const handleClearClick = () => {
+    setDisplay("0");
+    setSecondDisplay("0");
+    setResult("0");
+  };
+
+  const handleOperationClick = (op) => {
+    setOperation(op);
+    setSecondDisplay(display);
+    setDisplay("0");
+  };
+
+  const handleEqualsClick = () => {
+    switch (operation) {
+      case "+":
+        setResult(parseFloat(secondDisplay) + parseFloat(display));
+        break;
+      case "-":
+        setResult(parseFloat(secondDisplay) - parseFloat(display));
+        break;
+      case "*":
+        setResult(parseFloat(secondDisplay) * parseFloat(display));
+        break;
+      case "÷":
+        setResult(parseFloat(secondDisplay) / parseFloat(display));
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleStoreClick = () => {
+    setStoredValue(result);
+  };
+
+  const handleRecallClick = (displayNumber) => {
+    if (storedValue !== null) {
+      if (displayNumber === 1) {
+        setDisplay(storedValue);
+      } else if (displayNumber === 2) {
+        setSecondDisplay(storedValue);
+      }
+    }
+  };
 
   return (
     <div className="calculator">
       <div className="panel">
-        <p>0</p>
+        <p>{display}</p>
         <div className="numbers">
-          <button>1</button>
-          <button>2</button>
-          <button>3</button>
-          <button>4</button>
-          <button>5</button>
-          <button>6</button>
-          <button>7</button>
-          <button>8</button>
-          <button>9</button>
-          <button>0</button>
-          <button>Clear</button>
+          {[1, 2, 3, 4, 5, 6, 7, 8, 9, 0, "."].map((number, index) => (
```

Comment:
good, since `number` is already unique, you don't need `index` to set the `key`, just use number for the key